plugins/scheduled_task/lib/processors/KRecycleBinProcessor.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/**
+ * @package plugins.scheduledTask
+ * @subpackage lib.processors
+ */
+class KRecycleBinProcessor extends KGenericProcessor
+{
+	/**
+	 * @param KalturaScheduledTaskProfile $profile
+	 */
+	public function processProfile(KalturaScheduledTaskProfile $profile)
+	{
+		if ($this->wasHandledToday($profile->lastExecutionStartedAt))
+		{
+			KalturaLog::info("Recycle Bin Scheduled Task Profile [$profile->id] was already handled today. No need to handle again");
+			return;
+		}
+		
+		$this->taskRunner->impersonate($profile->partnerId);
+		try
+		{
+			$maxTotalCountAllowed = $this->preProcess($profile);
+			$objectsData = $this->handleProcess($profile, $maxTotalCountAllowed);
+			$this->postProcess($profile, $objectsData);
+			
+		} catch (Exception $ex)
+		{
+			$this->taskRunner->unimpersonate();
+			throw $ex;
+		}
+		$this->taskRunner->unimpersonate();
+	}
+	
+	/**
+	 * @param KalturaScheduledTaskProfile $profile
+	 * @param $maxTotalCountAllowed
+	 * @return mixed
+	 */
+	protected function handleProcess(KalturaScheduledTaskProfile $profile, $maxTotalCountAllowed)
+	{
+		$numberOfHandledEntries = 0;
+		$pageIndex = 1;
+		$daysBeforeDelete = $this->getPartnerDaysBeforeDelete($profile->partnerId);
+		$entriesList = $this->runSearch($pageIndex, $daysBeforeDelete);
+		while ($entriesList && $entriesList->totalCount);
+		{
+			$deleteResults = $this->deleteEntries($entriesList);
+			$numberOfHandledEntries += count($deleteResults);
+			$pageIndex++;
+			$entriesList = $this->runSearch($pageIndex, $daysBeforeDelete);
+		}
+		
+		KalturaLog::info("Number of recycled entries deleted for partner [{$profile->partnerId}]: $numberOfHandledEntries");
+	}
+	
+	/**
+	 * @param $partnerId
+	 * @return MultiRequestSubResult
+	 */
+	protected function getPartnerDaysBeforeDelete($partnerId)
+	{
+		KBatchBase::impersonate($partnerId);
+		$partner = KBatchBase::$kClient->partner->get($partnerId);
+		KBatchBase::unimpersonate();
+		return $partner->daysBeforeRecycleBinDeletion;
```

Comment:
if partner returned empty (and not with an error which will be caught from the upper try block) you will access null pointer
In this case we also need to throw an error as the number of days can falsely return as 0